projects/packages/autoloader/tests/php/bin/test-coverage.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * A utility for transforming a raw code coverage
+ * report into a clover.xml file for consumption.
+ *
+ * @package automattic/jetpack-autoloader
+ */
+
+use SebastianBergmann\CodeCoverage\Report\Clover;
+use SebastianBergmann\CodeCoverage\Version;
+
+// phpcs:disabled WordPress.Security.EscapeOutput.OutputNotEscaped
+
+define( 'ROOT_DIR', realpath( implode( DIRECTORY_SEPARATOR, array( __DIR__, '..', '..', '..' ) ) ) );
+
+require_once ROOT_DIR . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR . 'autoload.php';
+
+/**
+ * Returns the path to the clover.xml output file.
+ *
+ * @return string The path to the output file.
+ */
+function get_output_file() {
+	global $argc;
+	global $argv;
+
+	if ( $argc < 2 ) {
+		echo "Usage: test-coverage [clover.xml file]\n";
+		exit( -1 );
+	}
+
+	return $argv[1];
+}
+
+/**
+ * Attempts to load the report from the tmp file we should have generated.
+ *
+ * @return SebastianBergmann\CodeCoverage\CodeCoverage The unserialized code coverage object.
+ */
+function load_report() {
+	$coverage_report = implode( DIRECTORY_SEPARATOR, array( ROOT_DIR, 'tests', 'php', 'tmp', 'coverage-report.php' ) );
+	if ( ! file_exists( $coverage_report ) ) {
+		echo "There is no coverage report to process.\n";
+		exit( -1 );
+	}
+
+	return require_once $coverage_report;
+}
+
+/**
+ * Evaluates the version of sebastianbergmann/php-code-coverage that we've generated the coverage report using.
+ *
+ * @return string The version for the code coverage package.
+ */
+function get_coverage_version() {
+	return Version::id();
+}
+
+/**
+ * Counts the number of lines in a file before the `class` keyword.
+ *
+ * @param string $file The file to check.
+ * @return int|null The number of lines or null if there is no class keyword.
+ */
+function count_lines_before_class_keyword( $file ) {
+	// Find the line that the `class` keyword occurs on so that we can use it to calculate an offset from the header.
+	$content = file_get_contents( $file ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+
+	// Find the class keyword and capture the number of characters in the string before this point.
+	if ( 0 === preg_match_all( '/^class /m', $content, $matches, PREG_OFFSET_CAPTURE ) ) {
+		return null;
+	}
+
+	// Support both styles of line endings.
```

Comment:
The only case in which counting `\r\n` would differ from `\n` is if the file contains both kinds of line endings, which probably shouldn't happen.

And if it does, does the result from this code actually match what the coverage report reports? It doesn't on Linux; I don't have Windows available to test with.

Reply:
When I wrote this function it was with the intention of supporting mixed endings but it looks like I didn't follow through with it, thanks for catching that! While it's true that this shouldn't happen I wanted to write this script in such a way that it could have utility outside of this specific use-case.